server/app.js

```diff
@@ -9,39 +9,44 @@ import { createServer } from 'http';
 import { Model } from 'objection';
 import swaggerDocument from '@/swagger.json';
 import Routes from '@routes/v1';
+import dotenv from 'dotenv';
 import config from '@config';
 import webpackConfig from '@/webpack.config';
 import WebpackDevMiddleware from 'webpack-dev-middleware';
 import WebpackHotMiddleware from 'webpack-hot-middleware';
 
 import 'express-jsend';
 
+dotenv.config();
+
 export const app = express();
 
 export const databaseConnection = Knex(config.database[config.server.env]);
 
-const compiler = Webpack(webpackConfig);
-
-app.use(
-    WebpackDevMiddleware(compiler, {
-        hot: true,
-        publicPath: webpackConfig.output.publicPath
-    })
-);
+if (process.env.NODE_ENV == 'development') {
+    const compiler = Webpack(webpackConfig);
 
-app.use(WebpackHotMiddleware(compiler));
+    app.use(
+        WebpackDevMiddleware(compiler, {
+            hot: true,
+            publicPath: webpackConfig.output.publicPath
+        })
+    );
 
+    app.use(WebpackHotMiddleware(compiler));
+}
 Model.knex(databaseConnection);
 
 app.use(bodyParser.json());
 app.use(bodyParser.urlencoded({ extended: true }));
 
 Routes(app);
 
+const root = __dirname.substring(0, __dirname.lastIndexOf('/') + 1);
```

Comment:
I think you should use `path.resolve(__dirname, '..', 'public');`. Why did you perform the `root` computation? It looks risky and might not work on all environments.

Reply:
it removes everything after the last slash and appends the public name

Reply:
used yours